utils/src/check/super.c

```diff
@@ -241,8 +241,18 @@ int check_supers(int data_fd)
 	debug("super magic 0x%04x", global_super->hdr.magic);
 
 	debug("Superblock flag: %llu", global_super->flags);
-	if (global_super->flags != SCOUTFS_FLAG_IS_META_BDEV)
+	if (global_super->flags != SCOUTFS_FLAG_IS_META_BDEV) {
 		problem(PB_SB_BAD_FLAG, "Bad flag: %llu expecting: 1 or 0", global_super->flags);
+		if (repair) {
+			global_super->flags &= SCOUTFS_FLAG_IS_META_BDEV;
+			ret =super_commit();
+			if (ret < 0) {
+				fprintf(stderr, "error writing superblock\n");
+				goto out;
+			} else
+				debug("Superblock flag repaired: %llu", global_super->flags);
```

Comment:
You should call `correct(PB_SB_BAD_FLAG);` here to increment the counter for the corrected problem.

Reply:
Well yes, but, `correct()` already calls `debug()` in its own codepath. So you can remove the debug print out.